main/zoe/guide/contracts/nft-drop.md

```diff
@@ -0,0 +1,107 @@
+
+The purpose of the `NFTDropContractBasic.js` is to provide a point of reference for developers seeking to launch an NFT drop on the Agoric blockchain. While the implementation details are in complete control of the developer, some features that one may want to extend this contract with are:
+  * Fixing the total supply of NFTs available.
+  * Specifiying [the time](../../../repl/timerServices.md#timer-services):
+    * in which the NFT drop is open to the public.
+    * of an NFT drop deadline, after which users will no longer be able to mint and/or purchase.
+  * Embedded additional metadata relating the project.
+  * Whatever your heart desires!
+
+The code for the smart contract can be found below:
+```js
+/** @type {ContractStartFn} */
+const start = async (zcf) => {
+  const { pricePerNFT, nftName } = zcf.getTerms();
+
+  // Set up the mint
+  const mint = await zcf.makeZCFMint(nftName, AssetKind.SET);
+  const { brand: NFTBrand } = mint.getIssuerRecord();
+
+  let currentId = 1n;
+  const { zcfSeat: sellerSeat } = zcf.makeEmptySeatKit();
+
+  /** @type {OfferHandler} */
+  const buyNFTs = (buyerSeat) => {
+    // Mint the NFTs
+    const amount = AmountMath.make(NFTBrand, harden([currentId]));
+    mint.mintGains(harden({ NFTs: amount }), buyerSeat);
+    currentId += 1n;
+
+    // Take the money
+    sellerSeat.incrementBy(buyerSeat.decrementBy(harden({ Money: pricePerNFT })));
+
+    zcf.reallocate(buyerSeat, sellerSeat);
+    buyerSeat.exit();
+
+    return 'your offer was successful';
+  };
+
+  const publicFacet = Far('NFT Drop', {
+    makeInvitation: () => zcf.makeInvitation(buyNFTs, 'buyNFTs'),
+  });
+
+  return harden({ publicFacet });
+};
+
+```
+
+Taking a look inside of the contract code, you will notice that the minting process occurs **at the time of purchase**. This approach is in contrast to one in which a set of NFTs are minted prior to being made available to the public for purchase. 
+
+To accomplish this design, the contract first creates an internal `mint` variable. Note that the `mint` variable specifies the [`AssetKind`](../../../ertp/api/issuer.md#makeissuerkit-allegedname-assetkind-displayinfo) to be a `SET` which is necessary when creating NFTs with Zoe.
+
+```js
+const mint = await zcf.makeZCFMint(nftName, AssetKind.SET);
+```
+
+Once the `mint` has been created, the next step is to use it within the `buyNFTs` function which will be provided to the the public as an invitation. Additionally, we will see that each NFT has a unique identifier, `currentId`, that is embedded within the NFT at the time it is minted. Although it is intended to be a placeholder, the use of `currentId` is meant to demonstrate how one could add unique properties to an NFT.
```

Comment:
the `buyNFTs` function is distinct from an invitation; it's an offer handler associated with an invitation.

```suggestion
Once the `mint` has been created, the next step is to use it within the `buyNFTs` function which will be provided to the the public via an invitation. Additionally, we will see that each NFT has a unique identifier, `currentId`, that is embedded within the NFT at the time it is minted. Although it is intended to be a placeholder, the use of `currentId` is meant to demonstrate how one could add unique properties to an NFT.
```

Reply:
Yeah, i was teetering on whether to point out that `buyNFTs` function is a function but more specifically, it's an `offerHandler`. Will update though